File: AbstractFactoryDesignPattern/NotificationService_Example/Sender/EmailSendingService.java
package AbstractFactoryDesignPattern.NotificationService_Example.Sender;

import AbstractFactoryDesignPattern.NotificationService_Example.Message.Message;

public class EmailSendingService implements Sender {
	@Override
	public void sendMessage(Message message) {
		System.out.println("Sending message via SMTP: " + message.formatMessage());
	}
}

File: AbstractFactoryDesignPattern/NotificationService_Example_Python/SenderService/EmailSendingService.py
from AbstractFactoryDesignPattern.NotificationService_Example_Python.Message.NotificationMessage import \
	NotificationMessage
from AbstractFactoryDesignPattern.NotificationService_Example_Python.SenderService.NotificationService import \
	NotificationService


class EmailSendingService(NotificationService):

	def sendMessage(self, message: NotificationMessage) -> None:
		print(f"Sending Notification Message VIA SMTP: {message.formatMessage()}")
